docs/utilities/data_masking.md

```diff
@@ -0,0 +1,578 @@
+    --8<-- "examples/data_masking/src/generic_data_input.json"
+    ```
+
+=== "getting_started_mask_data_output.json"
+    ```json hl_lines="5 7 12"
+    --8<-- "examples/data_masking/src/getting_started_mask_data_output.json"
+    ```
+
+### Encrypting data
+
+???+ note "About static typing and encryption"
+    Encrypting data may lead to a different data type, as it always transforms into a string _(`<ciphertext>`)_.
+
+To encrypt, you will need an [encryption provider](#providers). Here, we will use `AWSEncryptionSDKProvider`.
+
+Under the hood, we delegate a [number of operations](#encrypt-operation-with-encryption-sdk-kms) to AWS Encryption SDK to authenticate, create a portable encryption message, and actual data encryption.
+
+=== "getting_started_encrypt_data.py"
+
+    ```python hl_lines="6-8 14-15 26"
+    --8<-- "examples/data_masking/src/getting_started_encrypt_data.py"
+    ```
+
+    1. You can use more than one KMS Key for higher availability but increased latency. </br></br>Encryption SDK will ensure the data key is encrypted with both keys.
+    2. See [working with nested data](#working-with-nested-data) to learn more about the `fields` parameter. </br></br>If we omit `fields` parameter, the entire dictionary will be encrypted.
+
+=== "generic_data_input.json"
+    ```json hl_lines="7-9 14"
+    --8<-- "examples/data_masking/src/generic_data_input.json"
+    ```
+
+=== "encrypt_data_output.json"
+    ```json hl_lines="5-7 12"
+    --8<-- "examples/data_masking/src/encrypt_data_output.json"
+    ```
+
+### Decrypting data
+
+???+ note "About static typing and decryption"
+    Decrypting data may lead to a different data type, as encrypted data is always a string _(`<ciphertext>`)_.
+
+To decrypt, you will need an [encryption provider](#providers). Here, we will use `AWSEncryptionSDKProvider`.
+
+Under the hood, we delegate a [number of operations](#decrypt-operation-with-encryption-sdk-kms) to AWS Encryption SDK to verify authentication, integrity, and actual ciphertext decryption.
+
+=== "getting_started_decrypt_data.py"
+
+    **NOTE**. Decryption only works with KMS Key ARN.
+
+    ```python hl_lines="6-7 12-13 24"
+    --8<-- "examples/data_masking/src/getting_started_decrypt_data.py"
+    ```
+
+    1. Note that KMS key alias or key ID won't work.
+    2. You can use more than one KMS Key for higher availability but increased latency. </br></br>Encryption SDK will call `Decrypt` API with all master keys when trying to decrypt the data key.
+    3. See [working with nested data](#working-with-nested-data) to learn more about the `fields` parameter.
+
+=== "encrypt_data_output.json"
+
+    ```json hl_lines="5-7 12"
+    --8<-- "examples/data_masking/src/encrypt_data_output.json"
+    ```
+
+=== "getting_started_decrypt_data_output.json"
+
+    ```json hl_lines="5-7 12-17"
+    --8<-- "examples/data_masking/src/getting_started_decrypt_data_output.json"
+    ```
+
+### Encryption context for integrity and authenticity
```

Comment:
I don't think AAD (the encryption context in this case) will help to prevent the confused deputy issue here right? Won't only proper authentication and authorization prevent that, especially since the encryption context is plaintext?